File: src/main/java/split/SplitStream.java
import split.data.LoremIpsum;

void main() {
    var result = splitStream(LoremIpsum.TEXT);
    println("result = " + result);
}

String splitStream(String line) {
    var pattern = Pattern.compile(" ");
    return pattern.splitAsStream(line)
            .filter(word -> word.length() > 3)
            .findFirst()
            .orElseThrow();
}

File: src/main/kotlin/splitkt/SplitStream.kt
package splitkt

import split.data.LoremIpsum

fun main() {
    val result = splitStream(LoremIpsum.TEXT)
    println(result)
}

fun splitStream(line: String): String =
    line.split(" ")
        .first { it.length > 3 } // orElseThrow() is implicit
